.github/workflows/go.yml

```diff
@@ -13,7 +13,45 @@ env:
   go-cache-name: go
 
 jobs:
+  changedfiles:
+    runs-on: [ubuntu-latest]
+    outputs:
+      docs: ${{ steps.changes.outputs.docs }}
+      code_changes: ${{ steps.changefiles.outputs.code_changes}}
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v3
+      with:
+        fetch-depth: 0
+    - name: Get changed files
+      id: changes
+      run: |
+        echo "::set-output name=docs::$(git diff --name-only ${{ github.event.before }}..${{ github.event.after }} | xargs)"
+    - name: echo output
+      id: changefiles
+      run: |
+        docs_changes=false
+        for file in ${{ steps.changes.outputs.docs }}; do
+             echo "changed files are $file"
+             ext="${file##*.}"
+             echo "$ext"
+             echo "$docs_changes"
+            if [[ ($ext != "md") && ($file != "hack/.md_links_config.json") ]]; then
```

Comment:
I think you also need to skip go jobs when there is only changes for files '*.yml'